dev_docs/tutorials/data/search.mdx

```diff
@@ -0,0 +1,490 @@
+      await ese.cancel(id, options, deps);
+    },
+    extend: async (id, keepAlive, options, deps) => {
+      // async search results are not stored indefinitely. By default, they expire after 7 days (or as defined by xpack.data_enhanced.search.sessions.defaultExpiration setting in kibana.yml).
+      // call the extend method of the async strategy you are using or implement your own extend function.
+      await ese.extend(id, options, deps);
+    },
+  };
+};
+```
+
+### High Level Search
+
+#### Search Source
+
+```ts
+function searchWithSearchSource() {
+  const indexPattern = data.indexPatterns.getDefault();
+  const query = data.query.queryString.getQuery();
+  const filters = data.query.filterManager.getFilters();
+  const timefilter = data.query.timefilter.timefilter.createFilter(indexPattern);
+  if (timefilter) {
+    filters.push(timefilter);
+  }
+
+  const searchSource = await data.search.searchSource.create();
+
+  searchSource
+    .setField('index', indexPattern)
+    .setField('filter', filters)
+    .setField('query', query)
+    .setField('fields', selectedFields.length ? selectedFields.map((f) => f.name) : ['*'])
+    .setField('aggs', getAggsDsl());
+    
+  searchSource.fetch$().subscribe({
+    next: () => {},
+    error: () => {},
+  });
+}
+```
+
+### Partial Results
+
+When searching using an `async` strategy (such as async DSL and async EQL), the search service will stream back partial results.   
+
+While you may ignore them and wait for the final result before rendering, you may also use those partial results to create a more interactive experience for your users. It is highly advised however, to make sure users are aware that the results they are seeing are partial.
+
+```ts
+// Handling partial results
+data.search.search(req).subscribe({
+  next: (result) => {
+    if (isCompleteResponse(res)) {
+      renderFinalResult(res);
+    } else if (isPartialResponse(res)) {
+      renderPartialResult(res);
+    }
+  },
+})
+
+// Skipping partial results
+const finalResult = await data.search.search(req).toPromise();
+```
+
+### Search Sessions
+
+A `Search Session` is a higher level concept, than search. It is used to describe a grouping of one or more async search requests with additional context. 
+
+`Search Sessions` come in handy when you want to enable a user to run something asynchronously (for example, a dashboard over a very long period of time), and then restore the results quickly at a later time, as the `Search Service` transparently fetches results from the `.async-search` index, instead of running each request again.
+
+Internally, any search run within the a `Search Session` is saved into a `Saved Object`, allowing `Kibana` to manage their lifecycle. Most `Saved Objects` are deleted automatically after a short period of time, but if a user chooses to save the `Search Session`, the `Saved Object` is persisted, so that results can be restored in a later time.
```

Comment:
```suggestion
Internally, any search run within a search session is saved into an object, allowing Kibana to manage their lifecycle. Most saved objects are deleted automatically after a short period of time, but if a user chooses to save the search session, the saved object is persisted, so that results can be restored in a later time.
```